### PR TITLE
[embind] Use `Object.assign` to setup prototypes. NFC

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

**Comment by RReverser** [comment]
Can we use real classes instead or do we still support any ancient browsers without ES6 support?

**Comment by sbc100** [comment]
> Can we use real classes instead or do we still support any ancient browsers without ES6 support?

No yet sadly.  We can currently only use ES6 features that clusure can lower away without helper code.    I hope this well change soon.

File: src/embind/embind.js [hunk 1]
```diff
@@ -1347,16 +1347,33 @@ var LibraryEmbind = {
...
+        return ptr;
+      },
+      destructor(ptr) {
+        if (this.rawDestructor) {
+          this.rawDestructor(ptr);
+        }
+      },
+      'argPackAdvance': GenericWireTypeSize,
+      'readValueFromPointer': readPointer,
+      'deleteObject': function(handle) {
```
Comment (RReverser) [inline src/embind/embind.js:1370]: FWIW if you want, and assuming Closure preserves them in the same way, you can also shorten such syntax here and in other places to

```js
'deleteObject'(handle) { ... }
````

Comment (sbc100) [reply src/embind/embind.js]: Done!

**Comment by sbc100** [comment]
Can one of you lgtm this?

**Review by RReverser — APPROVED** [review]

**Pull request merged**